src/rake.rs

```diff
@@ -77,24 +86,26 @@ impl Rake {
         word_score
     }
 
-    fn phrases<'a>(&'a self, text: &'a str) -> Vec<Vec<&'a str>> {
-        let mut phrases = Vec::new();
-        PUNC_RE.split(text).filter(|s| !s.is_empty()).for_each(|s| {
+    fn phrases<'a>(&self, phrases_iter: impl IntoIterator<Item = &'a str>) -> Vec<Vec<&'a str>> {
+        let phrases_iter = phrases_iter.into_iter();
+        let mut phrases = Vec::with_capacity(2 * phrases_iter.size_hint().0);
+        for s in phrases_iter.filter(|s| !s.is_empty()) {
```

Comment:
why did you replace `for_each`with `for` loop?

Reply:
It seems clearer, given that it's not a self-contained iterator, but rather the loop mutates the outer variable.